### PR TITLE
sshuttle: 0.71 -> 0.73

### PR DESCRIPTION
**Pull request by zimbatm**
couldn't test it on darwin though since it tries to pull-in iptables

### CODE REVIEW

**Comment by zimbatm** [comment]
do not merge, it still needs more testing and fixes

**Comment by nckx** [comment]
I'll test it this evening GMT.

**Comment by nckx** [comment]
Do iptables not exist on Darwin? Or are they not available for other reasons?

**Comment by nckx** [comment]
FYI: https://github.com/sshuttle/sshuttle/releases/tag/v0.74

**Comment by zimbatm** [comment]
It was actually more complicated because the new sshuttle is now a real python package so I had to shuffle things around. There was a bug in the setup.py which is now fixed in 0.74.

The latest push ^ now works for me on Darwin.

**Comment by nckx** [comment]
> @zimbatm: I had the homebrew version installed. I'm still getting the following issue:
> ```
> $ sshuttle -r modulo.zimbatm.com -v 0.0.0.0/0
> Starting sshuttle proxy.
> [local sudo] Password:
> <snip>
> ```

I'm going to hazard the guess that calling

```
$ sudo sshuttle -r modulo.zimbatm.com -v 0.0.0.0/0
```

directly works fine, because that's what happens here: `lib/python2.7/site-packages/sshuttle/client.py` tries to be — well, let's call it ‘clever’ by executing `sudo ${python} sshuttle` itself (line 167). And `${python}` can't find `shuttle` as well as the shell does.

The dirty hack of simply removing that line's `sys.executable, ` fixed that for me, although there's probably a nicer way to tell `${python}` where to search for its `argv[1]`.

File: pkgs/tools/security/sshuttle/default.nix [hunk 1]
```diff
@@ -1,55 +1,22 @@
...
-  installPhase = let
-    mapPath = f: x: stdenv.lib.concatStringsSep ":" (map f x);
-  in ''
-    mkdir -p $out/share/sshuttle
-    cp -R sshuttle *.py compat $out/share/sshuttle
-
-    mkdir -p $out/bin
-    ln -s $out/share/sshuttle/sshuttle $out/bin
-    wrapProgram $out/bin/sshuttle \
-      --prefix PATH : "${mapPath (x: "${x}/bin") buildInputs}" \
```
Comment (nckx) [inline pkgs/tools/security/sshuttle/default.nix]: Maybe `PYTHONPATH` can go now, but all this is quite deliberate.

With this gone, the user must manually install `iptables` (and make sure it remains installed, as it's no longer considered a run-time dependency) or face:

```
Starting sshuttle proxy.
firewall manager: Starting firewall with Python version 2.7.11
fatal: can't find either iptables or pfctl; check your PATH
```

Of course, by all means make it conditional on `isLinux` (and add `pfctl` if relevant).

Comment (nckx) [inline pkgs/tools/security/sshuttle/default.nix]: Or, alternatively, argue why it should be decoupled. :-)

Comment (zimbatm) [inline pkgs/tools/security/sshuttle/default.nix]: Fair, should be fixed in 573ec2d9ae732442766a887a88f33c88a4c63b18

**Comment by zimbatm** [comment]
`sys.executable` removed with a patch in 573ec2d9ae732442766a887a88f33c88a4c63b18, now it works for me on Darwin

**Comment by nckx** [comment]
Works for me on Linux, Travis too. @domenkozar, are you OK with the patchy hack?

**Comment by zimbatm** [comment]
Looks like upstream didn't like my patch: https://github.com/sshuttle/sshuttle/pull/57#issuecomment-170701957

**Comment by domenkozar** [comment]
Sure

**Pull request merged**